open_seq2seq/models/model.py

```diff
@@ -513,9 +525,9 @@ def compile(self, force_var_reuse=False, checkpoint=None, use_trt=False, precisi
         else:
           deco_print('Total trainable parameters: {}'.format(total_params))
 
-  def build_forward_pass_graph(self, input_tensors, gpu_id=0, checkpoint=None, use_trt=False, precision='FP32'):
+  def build_forward_pass_graph(self, input_tensors, gpu_id=0, checkpoint=None):
     """Wrapper around _build_forward_pass_graph with option of using TF-TRT"""
-    if use_trt:
+    if self.params.get("use_trt", False):
```

Comment:
I think it might be better if we refactor build_forward_pass_graph into two functions: build_tf_forward_pass_graph and build_trt_forward_pass_graph instead of the current form where we build the trt graph inside build_forward_pass_graph and the tensorflow graph inside _build_forward_pass_graph.

Reply:
I agree that the function names should be clearer, but I am hesitant to rename the abstract method `_build_forward_pass_graph`.

How about:
1. Remove `build_forward_pass_graph()`
2. Add `build_trt_forward_pass_graph()`
3. In `model.compile`, add the following:
```
if use_trt:
  build_trt_forward_pass_graph()
else:
  _build_forward_pass_graph()
```

Reply:
Good point. That seems like a good solution.

Reply:
thanks @trevor-m and @blisc !